src/seedu/addressbook/data/person/Address.java

```diff
@@ -1,19 +1,37 @@
 package seedu.addressbook.data.person;
 
 import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.address.AddressBlock;
+import seedu.addressbook.data.person.address.AddressPostalCode;
+import seedu.addressbook.data.person.address.AddressStreet;
+import seedu.addressbook.data.person.address.AddressUnit;
 
 /**
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
 
+    private static final int NUMBER_OF_PARAMETERS = 4;
     public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must be in 'a/BLOCK, STREET, UNIT, POSTAL_CODE' format.";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS_EXAMPLE = "a/123, Clementi Ave 3, #12-34, 231534";
+    public static final String ADDRESS_DELIMITER = ",";
+    public static final String ADDRESS_SKIP_IDENTIFIER = "a/";
 
-    public final String value;
+    private static final int ADDRESS_BLOCK_INDEX = 0;
+    private static final int ADDRESS_STREET_INDEX = 1;
+    private static final int ADDRESS_UNIT_INDEX = 2;
+    private static final int ADDRESS_POSTAL_CODE_INDEX = 3;
+
+    private static final int POSTAL_CODE_LENGTH = 6;
+
+    private final String value;
```

Comment:
Is there a real reason to keep value?
Since you have the address in 4 components, storing the address again would be duplication right?